src/scripting/lua_attributes.hpp

```diff
@@ -0,0 +1,162 @@
+/*
+	Copyright (C) 2009 - 2024
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+/// New attribute registration system, mainly for objects with a lot of attributes, like units
```

Comment:
"new" isn't a useful description in source code, for it will become "old that's still called new".